src/library/scala/collection/immutable/LongMap.scala

```diff
@@ -476,7 +476,7 @@ sealed abstract class LongMap[+T] extends AbstractMap[Long, T]
   override def ++ [V1 >: T](that: scala.collection.Iterable[(Long, V1)]): LongMap[V1] = concat(that)
 
   def collect[V2](pf: PartialFunction[(Long, T), (Long, V2)]): LongMap[V2] =
-    flatMap(kv => if (pf.isDefinedAt(kv)) new View.Single(pf(kv)) else View.Empty)
+    LongMap.from(new View.Collect(coll, pf))
```

Comment:
Why not builder-based implementations instead?

Reply:
Yes, builder-based would probably be better.

Reply:
I'll use https://github.com/scala/scala/pull/6776 (which I just merged) for this purpose.